src/actions/assignment.js

```diff
@@ -69,3 +89,37 @@ export function createAssignment(assignment, classroomId) {
     .catch(response => console.log('RESPONSE-error: ', response));
   };
 }
+
+export function fetchAssignments() {
+  console.log('-'.repeat(40), '\nfetchAssignments()\n');
+  return dispatch => {
+    dispatch({
+      type: types.REQUEST_ASSIGNMENTS,
+    });
+    return fetch(root + assignments, {
+      method: 'GET',
+      mode: 'cors',
+      headers: new Headers({
+          'Authorization': apiClient.headers.Authorization,
+          'Content-Type': 'application/json'
+        })
+      })
+      .then(response => response.json())
+      .then(json => {
+        console.log('!'.repeat(80), '\n', json.data);
+        dispatch({
```

Comment:
We should just send off the payload to the reducer, like this:

```
dispatch({
    type: types.RECEIVE_ASSIGNMENTS,
    payload: {
        assignments: json.data,
    },
});
```

It's the reducer's job to determine the rest of its state, like `error` and `loading`